File: javastory.budgetsh.stage3/src/javastory/budgetsh/stage3/budget/service/CashBookService.java
package javastory.budgetsh.stage3.budget.service;

import java.util.ArrayList;

import javastory.budgetsh.stage3.budget.entity.budget.CashBook;

public interface CashBookService {

	boolean exist(String bankAccount);

	boolean regist(CashBook cashbook);

	CashBook retrieve(String account);

	void update(CashBook foundCashbook);

	void remove(CashBook foundCashbook);

	ArrayList<CashBook> retrieveAll();

}

File: javastory.budgetsh.stage5.1/src/step1/logic/CashbookService.js
const CashBookStore = require('../store/CashBookStore.js');
const cashbookStore = new CashBookStore();

exports.exist = function(bankAccount) {
	if (cashbookStore.exist(bankAccount)) {
		return true;
	}
	return false;
}

exports.regist = function(cashbook) {
	let bankAccount = cashbook.bankAccount;
	if (cashbookStore.exist(bankAccount)) {
		return false;
	}
	cashbookStore.regist(cashbook);

	if (!cashbookStore.exist(bankAccount)) {
		return false;
	}
	return true;
}

exports.retrieve = function(bankAccount) {
	if (!cashbookStore.exist(bankAccount)) {
		return null;
	}
	return cashbookStore.retrieve(bankAccount);
}

exports.update = function(cashbook) {
	let bankAccount = cashbook.bankAccount;
	if (!cashbookStore.exist(bankAccount)) {
		return false;
	}
	cashbookStore.update(cashbook);
	return true;
}

exports.remove = function(cashbook) {
	let bankAccount = cashbook.bankAccount;
	if(!cashbookStore.exist(bankAccount)){
		return false;
	}
	cashbookStore.remove(cashbook);
	if(cashbookStore.exist(bankAccount)){
		return false;
	}
	return true;
}